cl_dll/tri.cpp

```diff
@@ -40,6 +60,91 @@ void CL_DLLEXPORT HUD_DrawNormalTriangles( void )
 void RunEventList( void );
 #endif
 
+void DivideRGBABy255(float &r, float &g, float &b, float &a)
+{
+	r /= 255.0f;
+	g /= 255.0f;
+	b /= 255.0f;
+	a /= 255.0f;
+}
+
+void DrawAACuboid(triangleapi_s *pTriAPI, Vector corner1, Vector corner2)
+{
+	pTriAPI->Begin(TRI_QUADS);
+
+	pTriAPI->Vertex3f(corner1.x, corner1.y, corner1.z);
+	pTriAPI->Vertex3f(corner1.x, corner2.y, corner1.z);
+	pTriAPI->Vertex3f(corner2.x, corner2.y, corner1.z);
+	pTriAPI->Vertex3f(corner2.x, corner1.y, corner1.z);
+
+	pTriAPI->Vertex3f(corner1.x, corner1.y, corner1.z);
+	pTriAPI->Vertex3f(corner1.x, corner1.y, corner2.z);
+	pTriAPI->Vertex3f(corner1.x, corner2.y, corner2.z);
+	pTriAPI->Vertex3f(corner1.x, corner2.y, corner1.z);
+
+	pTriAPI->Vertex3f(corner1.x, corner1.y, corner1.z);
+	pTriAPI->Vertex3f(corner2.x, corner1.y, corner1.z);
+	pTriAPI->Vertex3f(corner2.x, corner1.y, corner2.z);
+	pTriAPI->Vertex3f(corner1.x, corner1.y, corner2.z);
+
+	pTriAPI->Vertex3f(corner2.x, corner2.y, corner2.z);
+	pTriAPI->Vertex3f(corner1.x, corner2.y, corner2.z);
+	pTriAPI->Vertex3f(corner1.x, corner1.y, corner2.z);
+	pTriAPI->Vertex3f(corner2.x, corner1.y, corner2.z);
+
+	pTriAPI->Vertex3f(corner2.x, corner2.y, corner2.z);
+	pTriAPI->Vertex3f(corner2.x, corner1.y, corner2.z);
+	pTriAPI->Vertex3f(corner2.x, corner1.y, corner1.z);
+	pTriAPI->Vertex3f(corner2.x, corner2.y, corner1.z);
+
+	pTriAPI->Vertex3f(corner2.x, corner2.y, corner2.z);
+	pTriAPI->Vertex3f(corner2.x, corner2.y, corner1.z);
+	pTriAPI->Vertex3f(corner1.x, corner2.y, corner1.z);
+	pTriAPI->Vertex3f(corner1.x, corner2.y, corner2.z);
+
+	pTriAPI->End();
+}
+
+void DrawServerTriggers()
+{
+	if ((gHUD.m_pShowServerTriggers->value > 0) && (gHUD.m_pShowServerTriggers->value != 2.0f))
+	{
+		for (int e = 0; e < MAX_EDICTS; ++e)
+		{
+			cl_entity_t* ent = gEngfuncs.GetEntityByIndex(e);
+			if (ent)
+			{
+				if (ent->model)
+				{
+					if ((ent->curstate.rendermode == kRenderTransColor) && (ent->curstate.renderfx == kRenderFxTrigger))
+					{
+						color24 colors = ent->curstate.rendercolor;
+						if (!gHUD.IsTriggerForSinglePlayer(colors))
+						{
+							gEngfuncs.pTriAPI->RenderMode(kRenderTransAdd);
+							gEngfuncs.pTriAPI->CullFace(TRI_NONE);
+
+							float r = colors.r, g = colors.g, b = colors.b, a = std::min(255.0f, std::max(0.0f, gHUD.m_pShowServerTriggersAlpha->value));
+							DivideRGBABy255(r, g, b, a);
+							gEngfuncs.pTriAPI->Color4f(r, g, b, a);
+
+							Vector mins = ent->curstate.mins;
+							Vector maxs = ent->curstate.maxs;
+							Vector origin = ent->curstate.origin;
+							Vector absmin = origin + mins;
+							Vector absmax = origin + maxs;
+
+							DrawAACuboid(gEngfuncs.pTriAPI, absmin, absmax);
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+extern engine_studio_api_t IEngineStudio;
```

Comment:
Is this used anywhere?

Reply:
At first I used directly OpenGL functions and because of this I began to check whether we were using a hardware engine or not (`IEngineStudio.IsHardware()`), but later I abandoned this idea in favor of a solution that works optimally for both versions of engine

After that, I didn’t feel like to removing them, but I can do it you want

Reply:
Let's remove then this PR is good to go.